packages/engine/paima-sm/src/types.ts

```diff
@@ -521,6 +536,32 @@ export type ChainDataExtensionMinaActionGeneric = ChainDataExtensionBase &
     cdeType: ChainDataExtensionType.MinaActionGeneric;
   };
 
+export const ChainDataExtensionDynamicPrimitiveConfig = Type.Intersect([
+  ChainDataExtensionConfigBase,
+  Type.Object({
+    type: Type.Literal(CdeEntryTypeName.DynamicPrimitive),
+    targetType: Type.Literal(CdeEntryTypeName.ERC721),
```

Comment:
I'm not sure I understand why we have a target type instead of making the dynamic primitive "wrap" another arbitrary primitive type by nesting the YML config or something similar

Reply:
Well, my assumption was that you can't add the full config since some of the data may come in the event (like now with contract address). But maybe we can wrap the yml config making everything optional? The idea then would be to have a partial config here and then have a way to fill the rest from the contract event.

Reply:
Yeah I think the current setup you have with a dynamic NFT that fills in based on a `fields` map makes a lot of sense. It's not as clear to me:
1. Why we hardcoded this to only ERC721 (it feels like everything should work with other standards as-is)
2. The fact that it wraps is not immediate obvious from the shape of the configuration

I think your idea of making all the fields in the wrapped typed optional and checking that the fields included + the `fields` map + the implicit ones (contract address) gives the full configuration makes sense

Reply:
On the docs side it would probably help to have an annotate image to explain how the settings are for different categories (dynamic triggering settings, field settings, wrapped primitive settings)

Reply:
I moved the configs to a subfield already, it's now called `targetConfig`. I also renamed `fields` to `dynamicFields`, although I'm not sure if it's clear enough. It would look more like this now:

```yaml
extensions:
    - name: "dynamic nft"
...
    targetConfig: 
      scheduledPrefix: "nft"
      burnScheduledPrefix: "burn-nft"
      type: "erc721"
    dynamicFields:
      contractAddress: value
```

For some reason I can't use `Type.Partial` on the `ChainDataExtensionErc721Config` config though, I get a weird error: `Type instantiation is excessively deep and possibly infinite.`

As for this

> Why we hardcoded this to only ERC721 (it feels like everything should work with other standards as-is)

I wrote it with the idea of it being generic, I just reduced the scope to simplify testing and reviewing. But I can try extending this to the other ERC's. Testing every case would be super annoying though.

One issue is that I don't think it's possible to write the type to validate this correctly, but I can write a custom validation rule to check that `dynamicFields` + `config` provides a valid config.

Reply:
Okay, if leaving it to just ERC721 for now simplifies the type validation JS code we can leave it at that and create a Github issue explaining the limitation and the general idea on how to fix it (and then link to that issue from the docs)